### PR TITLE
OT-DeploySettingsMicroservice | Deployed settings microservice and database onto a Kubernetes cluster

### LINKED ISSUE #4: OT-DeploySettingsMicroservice
WHAT TO DO:
1. You will need to deploy this microservice onto Kubernetes. Here is a link to this repo: https://github.com/Off-Top-App/SettingsService

DONE: 
1. You are able to call both API's in this microservice when it is being served in kubernetes

### PR DESCRIPTION
**Pull request by frlzjosh**
**What I Did:**
1. Created several deployment scripts for the settings microservice as well as the database.
2. Ran these deployment scripts to deploy our microservice and database onto a Kubernetes cluster

**How To Test**
1. Follow this [**documentation**](https://docs.google.com/document/d/19_XEg7ZIM2NnJSQFBLUO1noKUih4YcKEMyRsV1iT0Is/edit) to set up a Kubernetes cluster through minikube
2. Follow this [**documentation**](https://docs.google.com/document/d/1cNMfzGiPlpmmegINQH_sEqL2pkWsgsMuvr8dOAFg4EM/edit) in order for you to deploy the settings microservice onto Kubernetes.
3. The documentation only has information on how to test the /ping endpoint, but in order for us to test if the database is also hosted, we will need to hit the POST request. Information on how to test that RESTful API can be found [**here**](https://docs.google.com/document/d/1cMh5mi0ZsyY7lW2JRwScSWB-uihMAF9s84shHdIIzMk/edit#bookmark=id.rq2wuh78pwve). 
   - **NOTE:** Make sure to use the URL provided when running `minikube service <service-name> --url` instead of `http://localhost:3000/` 
   - Make sure that you're able to get a response that looks like the one below. If so then you too have successfully deployed a microservice with its database onto Kubernetes.
```
{
    "success": 1,
    "data": {
        "app_color": "pink",
        "vibration_type": "medium",
        "default_categories": "[math, english, science]",
        "alert_type": "global",
        "user_email": "jdane@gmail.com"
    }
}
```

### CODE REVIEW

File: kubernetes-deployments/settings-db-secret.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+type: Opaque
+stringData:
+  APP_PORT: "3000"
+  DB_PORT: "3306"
+  DB_HOST: settings-db
+  DB_USER: root
+  DB_PASS: password
+  MYSQL_DB: userSettings
+  DB_DRIVER: mysql
+  MYSQL_ROOT_PASSWORD: password
```
Comment (tonynguyen98) [inline kubernetes-deployments/settings-db-secret.yml:14]: Remove these from DockerHub

**Review by tonynguyen98 — CHANGES_REQUESTED** [review]
Very good work! Was able to replicate the cluster on my machine. Just update the documentation to what was done during the meeting

File: kubernetes-deployments/settings-db-secret.yml (hunk 1 above)
Comment (frlzjosh) [reply kubernetes-deployments/settings-db-secret.yml]: done!

**Review by tonynguyen98 — APPROVED** [review]
Great work! I was able to replicate it on my machine after the changes

**Review by tonynguyen98 — APPROVED** [review]
Great work!

**Review by tonynguyen98 — APPROVED** [review]
Let's see if we can merge now

**Review by tonynguyen98 — APPROVED** [review]

**Pull request merged**